include/OpenSpaceToolkit/Astrodynamics/Trajectory/Sequence.hpp

```diff
@@ -0,0 +1,151 @@
+using ostk::astro::flight::system::dynamics::Thruster;
+
+/// @brief                      Represents a sequence of trajectory segments executed in order.
+
+class Sequence
+{
+   public:
+    struct Solution
+    {
+        Array<TrajectorySegment::Solution> segmentSolutions;
+
+        Array<State> getStates() const;
+    };
+
+    /// @brief                  Constructor
+    ///
+    /// @code
+    ///                         const Size repetitionCount = 1;
+    ///                         const NumericalSolver numericalSolver = NumericalSolver::Default();
+    ///                         const Array<Shared<Dynamics>> dynamicsArray =
+    ///                         {std::make_shared<CentralBodyGravity>(Earth::GravitationalParameter())};
+    ///                         const Duration maximumPropagationDuration = Duration::Days(7.0);
+    ///
+    ///                         Sequence sequence = {repetitionCount,numericalSolver, dynamicsArray,
+    ///                         maximumPropagationDuration};
+    ///
+    /// @endcode
+    ///
+    /// @param                  [in] aRepetitionCount A repetition count.
+    /// @param                  [in] aNumericalSolver A Numerical Solver. Defaults to Undefined.
+    /// @param                  [in] aDynamicsArray Optional array of shared dynamics.
+    /// @param                  [in] maximumPropagationDuration Maximum duration for propagation. Defaults to 7.0 days.
+
+    Sequence(
+        const Size& aRepetitionCount = 1,
+        const NumericalSolver& aNumericalSolver = NumericalSolver::Undefined(),
+        const Array<Shared<Dynamics>>& aDynamicsArray = Array<Shared<Dynamics>>::Empty(),
+        const Duration& maximumPropagationDuration = Duration::Days(7.0)
+    );
+
+    /// @brief                  Output stream operator.
+    ///
+    /// @param                  [in] anOutputStream An output stream.
+    /// @param                  [in] aSequence A sequence.
+    /// @return                 An output stream.
+
+    friend std::ostream& operator<<(std::ostream& anOutputStream, const Sequence& aSequence);
+
+    /// @brief                  Get segments.
+    ///
+    /// @return                 Array of segments.
+
+    Array<TrajectorySegment> getSegments() const;
+
+    /// @brief                  Get numerical solver.
+    ///
+    /// @return                 Numerical solver.
+
+    NumericalSolver getNumericalSolver() const;
+
+    /// @brief                  Get dynamics.
+    ///
+    /// @return                 Dynamics.
+
+    Array<Shared<Dynamics>> getDynamics() const;
+
+    /// @brief                  Get maximum propagation duration.
+    ///
+    /// @return                 Maximum propagation duration.
+
+    Duration getMaximumPropagationDuration() const;
+
+    /// @brief                  Add a trajectory segment.
+    ///
+    /// @param                  [in] aTrajectorySegment A trajectory segment.
+
+    void addSegment(const TrajectorySegment& aTrajectorySegment);
```

Comment:
Should they be Shared pointers? like the event condition below?

Reply:
Currently we only have a single Segment class, so I don't believe there's any value in having a shared pointer.